tezedge_state/src/proposer/proposal_persister.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::BufWriter;
+use std::sync::mpsc;
+
+use crate::proposals::*;
+
+pub struct ProposalPersisterHandle {
+    sender: Option<mpsc::SyncSender<RecordedProposal>>,
+    join_handle: Option<std::thread::JoinHandle<()>>,
+}
+
+impl Drop for ProposalPersisterHandle {
+    fn drop(&mut self) {
+        drop(self.sender.take());
+        if let Some(jh) = self.join_handle.take() {
+            let _ = jh.join();
+        }
+    }
+}
+
+impl Clone for ProposalPersisterHandle {
+    fn clone(&self) -> Self {
+        // TODO
+        unimplemented!("this is temporary, this type shouldn't be clonable")
+    }
+}
+
+impl ProposalPersisterHandle {
+    pub fn persist<P>(&mut self, proposal: P)
+    where
+        P: Into<RecordedProposal>,
+    {
+        // TODO: better error handling.
+        self.sender
+            .as_mut()
+            .unwrap()
+            .send(proposal.into())
+            .expect("recorded proposal persister disconnected!");
+    }
+}
+
+pub struct ProposalPersister {
+    channel: mpsc::Receiver<RecordedProposal>,
+}
+
+impl ProposalPersister {
+    pub fn start() -> ProposalPersisterHandle {
+        let (tx, rx) = mpsc::sync_channel(1024);
+
+        let state = Self { channel: rx };
+
+        let join_handle = std::thread::Builder::new()
+            .name("proposal-persister".to_owned())
+            .spawn(move || run(state))
+            .expect("failed to spawn proposal-persister thread");
```

Comment:
@binier 
please, change `expect` to propage error with Result instead,
I would leave that responsibility to caller,
caller should decide on error, if continue without storing proposals or if fail or propage error higher,

if this is called on starting of node, ok, then light/main.rs we can decide to crash the node immediatelly

Reply:
Yeah that is called just once when starting the node.